torchgeo/datasets/geo.py

```diff
@@ -307,12 +306,11 @@ def files(self) -> list[Path]:
         # Using set to remove any duplicates if directories are overlapping
         files: set[Path] = set()
         for path in paths:
-            if os.path.isdir(path):
-                pathname = os.path.join(path, '**', self.filename_glob)
-                files |= set(glob.iglob(pathname, recursive=True))
-            elif os.path.isfile(path) or path_is_vsi(path):
+            if os.path.exists(path) and os.path.isfile(path):
                 files.add(path)
-            elif not hasattr(self, 'download'):
+            else:
+                files |= set(list_directory_recursive(path, self.filename_glob))
+            if not files and not hasattr(self, 'download'):
```

Comment:
`not files` is going to become False as soon as any files are found. If you pass in 2 directories, 1 containing files and 1 containing nothing, the warning will not issue for the second directory.

Reply:
Thanks! I assigned the result of listing the current directory to a variable, and checked that, before warning/adding to set.